src/scope/scope.js

```diff
@@ -400,6 +400,35 @@ export default class Scope {
       });
   }
 
+  reset({ bitId, consumer }: { bitId: BitId, consumer?: Consumer}): Promise<consumerComponent> {
+    if (!bitId.isLocal(this.name)) {
+      return Promise.reject('you can not reset a remote component');
+    }
+    return this.sources.get(bitId)
+      .then((component) => {
+        if (!component) throw new ComponentNotFound(bitId.toString());
+        const allVersions = component.listVersions();
+        if (allVersions.length > 1) {
+          const lastVersion = component.latest();
+          bitId.version = lastVersion.toString();
+          return consumer.removeFromComponents(bitId, true).then(() => {
```

Comment:
> return consumer.removeFromComponents(bitId, true)

Maybe it will be better to separate the responsibilities in a way that the scope doesn't handle consumer operations.

We could put all the logic in `src/api/consumer/lib/reset.js`:

**scope.removeLastVersion(id)**
**consumer.removeFromComponents(id)**
**consumer.addToInlineComponents(id)**